target/cheri-common/cheri-translate-utils.h

```diff
@@ -258,20 +264,19 @@ static inline void gen_check_branch_target_dynamic(DisasContext *ctx, TCGv addr)
     //        call    raise_bounds_violation
     // Note: JR/JALR will often be used in hybrid/non-CHERI cases, so we can
     // skip the less than check if pcc.base is zero and top is MAX:
-    const bool need_base_check = ctx->base.pcc_base > 0;
-    const bool need_top_check = ctx->base.pcc_top != TYPE_MAXIMUM(target_ulong);
-    if (likely(!need_base_check && !need_top_check)) {
-        return;
+    if (likely(have_cheri_tb_flags(ctx, TB_FLAG_PCC_FULL_AS))) {
+        return; // PCC spans the full address space, no need to check
     }
+
     TCGLabel *skip_btarget_check = gen_new_label();
     TCGLabel *bounds_violation = gen_new_label();
     // We can skip the check of pcc.base if it is zero (common case in
     // hybrid/non-CHERI  mode).
-    if (unlikely(need_base_check)) {
+    if (unlikely(ctx->base.pcc_base > 0)) {
```

Comment:
Ditto re: `unlikely`?

Reply:
All hybrid/non-cheri code except for libcheri comparments runs with PCC.base zero so this should be likely. This function is only called for checking jalr/jr targets where the translator doesn't know the destination vaddr.